File: solutions/csharp/matrix/1/Matrix.cs
public class Matrix
{
    private int[][] _matrix;

    public Matrix(string input)
    {
        string[] rows = input.Split("\n");
        _matrix = new int[rows.Length][];
        for (int i = 0; i < rows.Length; i++)
        {
            string[] nums = rows[i].Split(" ");
            _matrix[i] = new int[nums.Length];
            for (int j = 0; j < nums.Length; j++)
            {
                int.TryParse(nums[j], out int num);
                _matrix[i][j] = num;
            }
        }
    }

    public int[] Row(int row) => _matrix[row - 1];

    public int[] Column(int col)
    {
        int[] column = new int[_matrix.Length];
        for (int i = 0; i < column.Length; i++)
            column[i] = _matrix[i][col - 1];
        return column;
    }
}

File: solutions/typescript/matrix/1/matrix.ts
export class Matrix {
  #matrix: number[][] = [];

  constructor(input: string) {
    const inputRows: string[] = input.split("\n");
    for (let i = 0; i < inputRows.length; i++) {
      const nums: string[] = inputRows[i].split(" ");
      const numbers: number[] = [];
      for (let num of nums)
        numbers.push(parseInt(num));
      this.#matrix.push(numbers);
    }
  }

  get rows(): number[][] {
    let matrixCopy: number[][] = [];
    for (let row of this.#matrix)
      matrixCopy.push([...row]);
    return matrixCopy;
  }

  get columns(): number[][] {
    let cols: number[][] = [];
    for (let i = 0; i < this.#matrix[0].length; i++)
      cols.push([]);

    for (let i = 0; i < this.#matrix.length; i++) {
      for (let j = 0; j < this.#matrix[i].length; j++) {
        cols[j].push(this.#matrix[i][j]);
      }
    }
    return cols;
  }
}
